modules/openapi-generator/src/main/resources/typescript-angular/api.service.mustache

```diff
@@ -67,6 +67,8 @@ export class {{classname}} {
         if (configuration) {
```

Comment:
wouldn't you want to set 
```
        if (configuration) {
            this.configuration = configuration;
            this.basePath = basePath || configuration.basePath || this.basePath;
        }
        this.configuration.basePath = this.basePath;
```
in order to account for the situation where you pass a `configuration` object that does not contain a base path plus a `basePath` parameter, since the `basePath` property of `Configuration` is optional, see
https://github.com/openapitools/openapi-generator/blob/master/modules/openapi-generator/src/main/resources/typescript-angular/configuration.mustache#L15 
?

Reply:
I would even mark the `basePath` property of the api service as deprecated, since only `this.configuration.basePath` should be used. what do you think about that?

Reply:
looks good now